src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -35,6 +35,12 @@ public class WalletSyncManager : IWalletSyncManager, IDisposable
         /// <summary>Queue which contains blocks that should be processed by <see cref="WalletManager"/>.</summary>
         private readonly AsyncQueue<Block> blocksQueue;
 
+        /// <summary>Limit blocksQueue size.</summary>
+        private const int MaxQueueSize = 1000;
```

Comment:
This is too much, this should be on the order of 100 or 200

Reply:
When I ran 500, 1000, 2000 blocks - when I was trying to setup a unit test, the blocks were being processed faster than the queue building up.  The queue count would stay around 0-1.

Admittedly the blocks have minimal txn's so not a real use case.  

Shall we bring it back down to 500 ?

Reply:
You need to think about the worst-case scenario when setting such limits. The worst-case scenario is the queue is full on bitcoin network when all blocks are very big. So let's assume you have 2 MB average block, then 100 limit here means 200 MB of memory that you need to have for this. I don't think it is reasonable to go to 500 or even thousand. I would go with just 100